packages/rocketchat-ui-flextab/client/tabs/messageSearch.js

```diff
@@ -10,11 +10,22 @@ Meteor.startup(function() {
 		],
 		action() {
 			const message = this._arguments[1];
+			if (Session.get('openedRoom') === message.rid) {
+				return RoomHistoryManager.getSurroundingMessages(message, 50);
+			}
+
+			FlowRouter.goToRoomById(message.rid);
+			// RocketChat.MessageAction.hideDropDown();
+
 			if (window.matchMedia('(max-width: 500px)').matches) {
 				Template.instance().tabBar.close();
 			}
 
-			RoomHistoryManager.getSurroundingMessages(message, 50);
+			window.setTimeout(() => {
+				RoomHistoryManager.getSurroundingMessages(message, 50);
```

Comment:
I see this is still there, I was trying so hard to find if there's any other way than setting a timeout, I can't remember if FlowRouter.goToRoomById returns any promise/callback